modules/drivers/google/src/metabase/driver/google.clj

```diff
@@ -70,46 +71,60 @@
 (defn- fetch-access-and-refresh-tokens* [scopes, ^String client-id, ^String client-secret, ^String auth-code]
   {:pre  [(seq client-id) (seq client-secret) (seq auth-code)]
    :post [(seq (:access-token %)) (seq (:refresh-token %))]}
+
   (log/info (u/format-color 'magenta (trs "Fetching Google access/refresh tokens with auth-code {0}..." (pr-str auth-code))))
   (let [^GoogleAuthorizationCodeFlow flow
         (.build (doto (GoogleAuthorizationCodeFlow$Builder. http-transport json-factory client-id client-secret scopes)
                   (.setAccessType "offline")))
 
-        ;; don't use `execute` here because this is a *different* type of Google request
+            ;; don't use `execute` here because this is a *different* type of Google request
         ^GoogleTokenResponse response
         (.execute (doto (.newTokenRequest flow auth-code)
                     (.setRedirectUri redirect-uri)))]
     {:access-token (.getAccessToken response), :refresh-token (.getRefreshToken response)}))
 
-(def ^{:arglists '([scopes client-id client-secret auth-code])} fetch-access-and-refresh-tokens
+(def ^{:arglists '([scopes client-id client-secret auth-code service-account-json])} fetch-access-and-refresh-tokens
```

Comment:
@robdaemon it looks like while `fetch-access-and-refresh-tokens` passes `service-account-json` through to `fetch-access-and-refresh-tokens*`, the unmemoized function doesn't accept that arity.

What should happen? Should `fetch-access-and-refresh-tokens*` accept `service-account-json` but ignore it?

Reply:
ugh that is totally a bug on my part. `fetch-access-and-refresh-tokens` doesn't need `service-account-json` at all, it won't act on it.